schema/testing.js

```diff
@@ -22,15 +22,74 @@ module.exports = (fastify, opts, next) => {
     const [test] = await Test.query().where({ id: query.test_id });
     return { result: test.getUserResult(session.userid) };
   });
-  fastify.post('/answer', async ({ query, session }) => {
-    const [question] = await Question.query().where({ id: query.question_id });
+  fastify.post('/answer', async ({ body, session }) => {
+    const [question] = await Question.query().where({ id: body.question_id });
     assert(
-      await question.validateAnswers(query.answer_ids),
+      await question.validateAnswers(body.answer_ids),
       new APIError(400, 'Some of the answers is invalid')
     );
-    const rows = query.answer_ids.map(id => ({ user_id: session.userid, answer_id: id }));
+    const rows = body.answer_ids.map(id => ({ user_id: session.userid, answer_id: id }));
     await Progress.query().insertGraph(rows);
     return { answer: 'Ok' };
   });
+
+  fastify.post('/categories/add', async ({ body, session }) => {
+    const [categories] = await Category.query().where({ name: body.name });
+    assert(!categories, new APIError(400, 'Not added test'));
```

Comment:
```suggestion
    assert(!categories, new APIError(409, 'This name is busy'));
```

Reply:
Also, why is it called `tests` if it only describes an item, not the list of items? The same appeals to the lines below.

Reply:
Maybe "This category already exists"?